package/src/components/Chat/Chat.tsx

```diff
@@ -204,14 +204,21 @@ const ChatWithContext = <
 
   useEffect(() => {
     if (userID && enableOfflineSupport) {
+      // This acts as a lock for some very rare occurrences of concurrency
+      // issues we've encountered before with the QuickSqliteClient being
+      // uninitialized before it's being invoked.
       setInitialisedDatabaseConfig({ initialised: false, userID });
       QuickSqliteClient.initializeDatabase();
-      DBSyncManager.init(client as unknown as StreamChat);
       setInitialisedDatabaseConfig({ initialised: true, userID });
+      DBSyncManager.init(client as unknown as StreamChat);
```

Comment:
Moved this down for more clarity as discussed with @santhoshvai  - the lock is there for `QuickSqliteClient.initializeDatabase()` and not for `DBSyncManager.init()` in particular. This way it should be more clear.